src/common/syncjournaldb.cpp

```diff
@@ -1028,6 +1050,114 @@ bool SyncJournalDb::getFileRecord(const QString &filename, SyncJournalFileRecord
     return true;
 }
 
+bool SyncJournalDb::getFileRecordByInode(quint64 inode, SyncJournalFileRecord *rec)
+{
+    QMutexLocker locker(&_mutex);
+
+    // Reset the output var in case the caller is reusing it.
+    Q_ASSERT(rec);
+    rec->_path.clear();
+    Q_ASSERT(!rec->isValid());
+
+    if (!inode)
+        return true; // no error, yet nothing found (rec->isValid() == false)
+
+    if (!checkConnect())
+        return false;
+
+    _getFileRecordQueryByInode->reset_and_clear_bindings();
+    _getFileRecordQueryByInode->bindValue(1, inode);
+
+    if (!_getFileRecordQueryByInode->exec()) {
+        return false;
+    }
+
+    if (_getFileRecordQueryByInode->next()) {
+        rec->_path = _getFileRecordQueryByInode->baValue(0);
+        rec->_inode = _getFileRecordQueryByInode->intValue(1);
+        rec->_modtime = _getFileRecordQueryByInode->int64Value(2);
+        rec->_type = _getFileRecordQueryByInode->intValue(3);
+        rec->_etag = _getFileRecordQueryByInode->baValue(4);
+        rec->_fileId = _getFileRecordQueryByInode->baValue(5);
+        rec->_remotePerm = _getFileRecordQueryByInode->baValue(6);
+        rec->_fileSize = _getFileRecordQueryByInode->int64Value(7);
+        rec->_serverHasIgnoredFiles = (_getFileRecordQueryByInode->intValue(8) > 0);
+        rec->_checksumHeader = _getFileRecordQueryByInode->baValue(9);
+    }
+
+    return true;
+}
+
+bool SyncJournalDb::getFileRecordByFileId(const QByteArray &fileId, SyncJournalFileRecord *rec)
+{
+    QMutexLocker locker(&_mutex);
+
+    // Reset the output var in case the caller is reusing it.
+    Q_ASSERT(rec);
+    rec->_path.clear();
+    Q_ASSERT(!rec->isValid());
+
+    if (fileId.isEmpty())
+        return true; // no error, yet nothing found (rec->isValid() == false)
+
+    if (!checkConnect())
+        return false;
+
+    _getFileRecordQueryByFileId->reset_and_clear_bindings();
+    _getFileRecordQueryByFileId->bindValue(1, fileId);
+
+    if (!_getFileRecordQueryByFileId->exec()) {
+        return false;
+    }
+
+    if (_getFileRecordQueryByFileId->next()) {
+        rec->_path = _getFileRecordQueryByFileId->baValue(0);
+        rec->_inode = _getFileRecordQueryByFileId->intValue(1);
+        rec->_modtime = _getFileRecordQueryByFileId->int64Value(2);
+        rec->_type = _getFileRecordQueryByFileId->intValue(3);
+        rec->_etag = _getFileRecordQueryByFileId->baValue(4);
+        rec->_fileId = _getFileRecordQueryByFileId->baValue(5);
+        rec->_remotePerm = _getFileRecordQueryByFileId->baValue(6);
+        rec->_fileSize = _getFileRecordQueryByFileId->int64Value(7);
+        rec->_serverHasIgnoredFiles = (_getFileRecordQueryByFileId->intValue(8) > 0);
+        rec->_checksumHeader = _getFileRecordQueryByFileId->baValue(9);
```

Comment:
Could this data extraction also be a common function, preferably defined close the the query string macro?

Reply:
done, it indeed looks cleaner